File: 0974-reorder-data-in-log-files/0974-reorder-data-in-log-files.kt
class Solution {
    fun reorderLogFiles(logs: Array<String>): Array<String> {
        val letters = mutableListOf<String>()
        val digits = mutableListOf<String>()
        
        for (log in logs) {
            if (Character.isDigit(log.split(" ")[1][0])) {
                digits.add(log)
            } else {
                letters.add(log)
            }
        }

        letters.sortWith(Comparator { s1: String, s2: String ->
            val s1x = s1.split(" ", limit = 2)
            val s2x = s2.split(" ", limit = 2)

            val compared = s1x[1].compareTo(s2x[1])
            if (compared == 0) {
                s1x[0].compareTo(s2x[0])
            } else {
                compared
            }
        })

        letters.addAll(digits)

        return letters.toTypedArray()
    }
}

File: 0974-reorder-data-in-log-files/0974-reorder-data-in-log-files.py
class Solution:
    def reorderLogFiles(self, logs: List[str]) -> List[str]:
        letters = []
        digits = []

        for log in logs:
            parts = log.split(" ", 1)
            if parts[1][0].isdigit():
                digits.append(log)
            else:
                letters.append(log)

        letters.sort(key=lambda log: (log.split(" ", 1)[1], log.split(" ", 1)[0]))

        return letters + digits
        